### PR TITLE
feat: expandString function

### PR DESCRIPTION
**Pull request by sharmaigne**
Expands valid string inputs. If the input is invalid, returns the string "Invalid Input"

Tested with a few cases:
VALID
simple "2aAb3c"
multidigit prefixes "10D3a"

INVALID
has space/non-alphanumeric "2 a 4b " "3$4b"
ends with number "ab2"

### CODE REVIEW

File: Mabano_Lada_PE02.cpp [hunk 1]
```diff
@@ -127,7 +131,26 @@ void compressionLoop() {
 }
 
 string expandString(string s) {
+  /* valid:  */
+  string invalid = "Invalid Input";
```
Comment (44mira) [inline Mabano_Lada_PE02.cpp:135]: use define

File: Mabano_Lada_PE02.cpp [hunk 2]
```diff
@@ -81,10 +81,14 @@ void expansionLoop() {
...
 
   while (true) {
-    cout << "Enter a compressed string: ";
+    cout << "\nEnter a compressed string: ";
     getline(cin, s);
-
-    cout << "\nExpanded form: " << expandString(s);
+    
+    string result = expandString(s);
+    if (result == "invalid input")
```
Comment (44mira) [inline Mabano_Lada_PE02.cpp]: use define

**Review by 44mira — CHANGES_REQUESTED** [review]
use a define for invalid

File: Mabano_Lada_PE02.cpp [hunk 3]
```diff
@@ -127,7 +131,26 @@ void compressionLoop() {
...
+    /* while digit, collect into prefixNum */
+    while (isdigit(s[i]))
+      prefixNum += s[i++];
+    
+    /* i >= s.size() : if gets here if the string ends with a digit (invalid)
+    For n=1, the value of n SHOULD NOT be shown (?) */
+    if (i >= s.size() || !isalpha(s[i]) || prefixNum == "1")
+      return invalid;
+
+    expandedString += std::string( (prefixNum == "") ? 1 : stoi(prefixNum), s[i] );
```
Comment (44mira) [inline Mabano_Lada_PE02.cpp]: avoid use of constructors

File: Mabano_Lada_PE02.cpp (hunk 2 above)
Comment (sharmaigne) [reply Mabano_Lada_PE02.cpp]: man na boss

**Review by 44mira — APPROVED** [review]

**Pull request merged**